OpenHPL/Waterway/SurgeTanks.mo

```diff
@@ -1,33 +1,38 @@
-within OpenHPL.Waterway;
-model SurgeTankAirCushion "Model of air-cushion surge tank"
+﻿within OpenHPL.Waterway;
+model SurgeTanks "Model of surge tanks"
   outer Parameters para "Parameters";
   extends OpenHPL.Icons.Surge;
   import Modelica.Constants.pi;
-  //// geometrical parameters of air-cushion surge tank
+  // Geometrical parameters of for surge tanks
   parameter Modelica.SIunits.Height H = 10.5 "Vertical component of the length of the surge shaft" annotation (
     Dialog(group = "Geometry"));
+  parameter Types.SurgeTank surge_tank_type = OpenHPL.Types.SurgeTank.SimpleST "Types of surge tank" annotation (
+    Dialog(group = "Surge tank types"));
   parameter Modelica.SIunits.Length L = 12.12 "Length of the surge shaft" annotation (
     Dialog(group = "Geometry"));
   parameter Modelica.SIunits.Diameter D = 50.47 "Diameter of the surge shaft" annotation (
     Dialog(group = "Geometry"));
+  parameter Modelica.SIunits.Diameter D_so = 25 "If Orifice type: Diameter of sharp orifice" annotation (
+    Dialog(group = "Geometry",enable=surge_tank_type == OpenHPL.Types.SurgeTank.SharpOrificeST));
+
   parameter Modelica.SIunits.Height eps = para.eps "Pipe roughness height" annotation (
     Dialog(group = "Geometry"));
-  //// condition for steady state
+  // Condition for steady state
   parameter Boolean SteadyState = para.Steady "If true - starts from Steady State" annotation (
     Dialog(group = "Initialization"));
-  //// steady state values for flow rate and water level in surge tank
+  // steady state values for flow rate and water level in surge tank
   parameter Modelica.SIunits.VolumeFlowRate V_dot0 = 0 "Initial flow rate in the surge tank" annotation (
     Dialog(group = "Initialization"));
   parameter Modelica.SIunits.Height h_0 = 2 "Initial water height in the surge tank" annotation (
     Dialog(group = "Initialization"));
-  input Modelica.SIunits.Pressure p_c0 = 4*para.p_a "Initial pressure of air-cushion inside the surge tank" annotation (
-    Dialog(group = "Geometry"));
+  input Modelica.SIunits.Pressure p_t0 = 4*para.p_a "Initial pressure of air-cushion inside the surge tank" annotation (
```

Comment:
The p_t0 should simple be set to a value. Even for normal surge tanks one might want to specify a different pressure because of altitude. Why do you make this an input? A simple parameter would be fine and does not require an additional input qualifier.